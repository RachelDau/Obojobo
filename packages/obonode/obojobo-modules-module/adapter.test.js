jest.mock('obojobo-document-engine/src/scripts/common/models/obo-model', () => {
	return require('obojobo-document-engine/__mocks__/obo-model-adapter-mock').default
})
import OboModel from 'obojobo-document-engine/src/scripts/common/models/obo-model'

import ModuleAdapter from './adapter'

describe('Module adapter', () => {
	beforeEach(() => {
		jest.resetAllMocks()
	})

	test('construct builds without attributes', () => {
		const model = new OboModel({})

		ModuleAdapter.construct(model)
		expect(model.modelState).toMatchSnapshot()
	})

	test('construct builds with attributes', () => {
		const attrs = {
			content: {
				start: 'mockStart'
			}
		}
		const model = new OboModel(attrs)

		ModuleAdapter.construct(model, attrs)
		expect(model.modelState).toMatchSnapshot()
	})

	test('clone creates a copy', () => {
		const a = {
			modelState: {
				start: 'mockStart'
			}
		}
		const b = { modelState: {} }

		ModuleAdapter.clone(a, b)

		expect(a).not.toBe(b)
		expect(a).toEqual(b)
	})

	test('toJSON builds a JSON representation', () => {
		const a = {
			modelState: {
				start: 'mockStart'
			}
		}
		const b = { content: {} }

		ModuleAdapter.toJSON(a, b)

		expect(a).not.toBe(b)
		expect(a.modelState.start).toEqual(b.content.start)
	})
})
