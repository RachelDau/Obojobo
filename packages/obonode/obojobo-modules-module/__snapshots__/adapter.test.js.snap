// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Module adapter construct builds with attributes 1`] = `
Object {
  "start": "mockStart",
}
`;

exports[`Module adapter construct builds without attributes 1`] = `
Object {
  "start": null,
}
`;
